src/main/java/build/buildfarm/cas/CASFileCache.java

```diff
@@ -177,6 +178,7 @@ public void onRemoval(
 
   private transient long sizeInBytes = 0;
   private transient Entry header = new SentinelEntry();
+  private final AtomicLong unreferencedEntryCount = new AtomicLong(0);
```

Comment:
Do you need an AtomicLong here? A volatile might suffice, as all list operations are already guarded by `synchronized(this)`